src/utils/vector.c

```diff
@@ -109,14 +109,26 @@ int vector_delete_at(vector *vec, unsigned index) {
     }
 
     // We deleted an entry, so blocks-1
-    if(vec->blocks > 0) {
-        vec->blocks--;
-    }
+    vec->blocks--;
 
     // Return success
     return 0;
 }
 
+int vector_swapdelete_at(vector *vec, unsigned index) {
```

Comment:
The name feels a bit, but I couldn't come up with anything better. I considered making this the default, but I think we rely on vector keeping the order in most places. Blah.